### PR TITLE
LCORE-1826: Expose OpenTelemetry configuration in v1/config endpoint

### PR DESCRIPTION
**Pull request by anik120**
## Description

Add observability configuration to the v1/config endpoint, exposing all OTEL_* environment variables to provide visibility into the active OpenTelemetry tracing setup.

## Type of change

- [ ] Refactor
- [ ] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up service version
- [ ] Bump-up dependent library
- [ ] Bump-up library or tool used for development (does not change the final image)
- [ ] CI configuration change
- [ ] Konflux configuration change
- [ ] Unit tests improvement
- [ ] Integration tests improvement
- [ ] End to end tests improvement
- [ ] Benchmarks improvement

## Tools used to create PR

Identify any AI code assistants used in this PR (for transparency and review context)

- Assisted-by: (e.g., Claude, CodeRabbit, Ollama, etc., N/A if not used)
- Generated by: (e.g., tool name and version; N/A if not used)

## Related Tickets & Documents

- Related Issue #
- Closes #

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] PR has passed all pre-merge test jobs.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Please provide detailed steps to perform tests related to this code change.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.

## Summary by CodeRabbit

* **New Features**
  * Added an `observability` section to the configuration response.
  * Automatically surfaces OpenTelemetry settings from `OTEL_*` environment variables.
  * Redacts sensitive OpenTelemetry values (headers and certificate/key-related fields) before exposing them.
  * Updated the published API documentation with the new observability configuration schema.

* **Tests**
  * Added integration tests validating `/config` includes observability and correctly collects `OTEL_*` values.
  * Added unit tests covering defaults, parsing, validation, and secret redaction edge cases.
  * Updated configuration dump expectations to include `observability`.

### CODE REVIEW

File: src/models/config.py [hunk 1]
```diff
@@ -2772,6 +2831,38 @@ def compiled_patterns(self) -> CompiledPatterns:
...
+        """Collect all OTEL_* environment variables from the environment.
+
+        Returns:
+            ObservabilityConfiguration with otel dict populated from environment.
+        """
+        otel_vars = {}
+        for key, value in os.environ.items():
+            if key.startswith("OTEL_"):
+                # Exclude sensitive variables that might contain tokens or keys
+                # Currently no secret OTEL_ vars are used, but this is future-proof
```
Comment (asimurka) [inline src/models/config.py]: This is actually not true. Anyone can export optional env variables and those will be leaked now.
Please research which OTEL variables can contain sensitive values and wrap those into `SecretStr` for example. The list of all configurable OTEL variables is attached in design doc

**Review by asimurka — CHANGES_REQUESTED** [review]

File: src/models/config.py [hunk 2]
```diff
@@ -2772,6 +2831,83 @@ def compiled_patterns(self) -> CompiledPatterns:
...
+        # Create a new dict to avoid mutating caller's data
+        redacted = {}
+
+        for key, val in value.items():
+            # Redact generic and signal-specific OTLP headers
+            # Matches: OTEL_EXPORTER_OTLP_HEADERS,
+            #          OTEL_EXPORTER_OTLP_TRACES_HEADERS,
+            #          OTEL_EXPORTER_OTLP_METRICS_HEADERS,
+            #          OTEL_EXPORTER_OTLP_LOGS_HEADERS
+            if "HEADERS" in key and "OTLP" in key:
```
Comment (asimurka) [inline src/models/config.py:2882]: If multiple headers are present, they are specified in the following format:
```bash
export OTEL_EXPORTER_OTLP_HEADERS=\
  "api-key=YOUR_SECRET_TOKEN,tenant-id=acme"
```
Please keep key names visible and redact only values

**Review by asimurka — CHANGES_REQUESTED** [review]

**Review by asimurka — APPROVED** [review]
LGTM

**Review by tisnik — APPROVED** [review]
LGTM
